test/quantization/test_quantize_fx.py

```diff
@@ -168,6 +173,51 @@ def forward(self, x):
         ref_qparams = weight_obs.calculate_qparams()
         self.assertEqual(qparams, ref_qparams)
 
+    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    @override_qengines
+    def test_qat_prepare_device_affinity(self):
+        """
+        Tests that FX QAT prepare pass respects device affinity
+        """
+        class Model(nn.Module):
+
+            def __init__(self):
+                super(Model, self).__init__()
+                self.conv = nn.Conv2d(1, 1, 1)
+                self.bn = nn.BatchNorm2d(1)
+                self.relu = nn.ReLU()
+
+            def forward(self, x):
+                x = self.conv(x)
+                x = self.bn(x)
+                x = self.relu(x)
+                return x
+
+        model = Model()
+        qengine = torch.backends.quantized.engine
+        qconfig_dict = {'': torch.quantization.get_default_qat_qconfig(qengine)}
+        device = torch.device('cuda:0')
+        model.to(device)
+
+        # symbolically trace
+        model = symbolic_trace(model)
+
+        # QAT prepare
+        model = fuse_fx(model)
+        model = prepare_fx(model, qconfig_dict)
```

Comment:
Not related to this PR, but do we plan to change this to `prepare_fx_qat` or something in the future?

Reply:
not sure, would be great to finalize the APIs we want as a team this month!